doc/source/conf.py

```diff
@@ -332,126 +182,40 @@
 # Numpy extensions
 # -----------------------------------------------------------------------------
 numpydoc_show_class_members = False
 numpydoc_class_members_toctree = False
 
-# -----------------------------------------------------------------------------
-# Plots
-# -----------------------------------------------------------------------------
-plot_basedir = os.path.join(curpath, "plots")
-plot_pre_code = """
-import numpy as np
-import matplotlib.pyplot as plt
-
-import matplotlib
-matplotlib.rcParams.update({
-    'font.size': 14,
-    'axes.titlesize': 12,
-    'axes.labelsize': 10,
-    'xtick.labelsize': 8,
-    'ytick.labelsize': 8,
-    'legend.fontsize': 10,
-    'figure.subplot.bottom': 0.2,
-    'figure.subplot.left': 0.2,
-    'figure.subplot.right': 0.9,
-    'figure.subplot.top': 0.85,
-    'figure.subplot.wspace': 0.4,
-    'text.usetex': False,
-})
-
-"""
-plot_include_source = True
-plot_formats = [('png', 100), ('pdf', 100)]
-
-plot2rst_index_name = 'README'
-plot2rst_rcparams = {'image.cmap' : 'gray',
-                     'image.interpolation' : 'none'}
-
 # -----------------------------------------------------------------------------
 # intersphinx
 # -----------------------------------------------------------------------------
-_python_version_str = f'{sys.version_info.major}.{sys.version_info.minor}'
-_python_doc_base = 'https://docs.python.org/' + _python_version_str
+_python_version_str = f"{sys.version_info.major}.{sys.version_info.minor}"
+_python_doc_base = "https://docs.python.org/" + _python_version_str
 intersphinx_mapping = {
-    'python': (_python_doc_base, None),
-    'numpy': ('https://numpy.org/doc/stable',
-              (None, './_intersphinx/numpy-objects.inv')),
-    'scipy': ('https://docs.scipy.org/doc/scipy/',
-              (None, './_intersphinx/scipy-objects.inv')),
-    'sklearn': ('https://scikit-learn.org/stable',
-                (None, './_intersphinx/sklearn-objects.inv')),
-    'matplotlib': ('https://matplotlib.org/',
-                   (None, './_intersphinx/matplotlib-objects.inv'))
+    "python": (_python_doc_base, None),
+    "numpy": (
+        "https://numpy.org/doc/stable/",
+        (None, "./_intersphinx/numpy-objects.inv"),
+    ),
+    "scipy": (
+        "https://docs.scipy.org/doc/scipy/",
+        (None, "./_intersphinx/scipy-objects.inv"),
+    ),
+    "sklearn": (
+        "https://scikit-learn.org/stable/",
+        (None, "./_intersphinx/sklearn-objects.inv"),
+    ),
+    "matplotlib": (
+        "https://matplotlib.org/stable/",
+        (None, "./_intersphinx/matplotlib-objects.inv"),
+    ),
 }
 
-# ----------------------------------------------------------------------------
-# Source code links
-# ----------------------------------------------------------------------------
-
-import inspect
-from os.path import relpath, dirname
-
-
-# Function courtesy of NumPy to return URLs containing line numbers
-def linkcode_resolve(domain, info):
```

Comment:
You replaced [sphinx.ext.linkcode](https://www.sphinx-doc.org/en/master/usage/extensions/linkcode.html) with [sphinx.ext.viewcode](https://www.sphinx-doc.org/en/master/usage/extensions/viewcode.html)? I can see that this reduces the complexity a bit because this hook is no longer necessary. 

If that is the only reason though, I'm in favor of reverting to linkcode. Especially if it is already working fine. I find that feature quite frequently very useful when looking up stuff during review. And I'd say GitHub provides a better UI, more context and tools compared to viewcode.

Reply:
@stefanv also prefers linkcode over viewcode (see 3c42f4f).